packages/kbn-shared-ux-components/src/page_template/page_template_inner.tsx

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { FunctionComponent } from 'react';
+
+import { EuiEmptyPrompt, EuiPageTemplate } from '@elastic/eui';
+import { withSolutionNav } from './with_solution_nav';
+import { KibanaPageTemplateProps } from './types';
+import { getClasses } from './util';
+
+type Props = KibanaPageTemplateProps;
+
+/**
+ * A thin wrapper around EuiPageTemplate with a few Kibana specific additions
+ */
+export const KibanaPageTemplateInner: FunctionComponent<Props> = ({
+  template,
+  className,
+  pageHeader,
+  children,
+  isEmptyState,
+  ...rest
+}) => {
+  /**
+   * An easy way to create the right content for empty pages
+   */
+  const emptyStateDefaultTemplate = 'centeredBody';
+  let header = pageHeader;
+  if (isEmptyState) {
+    if (pageHeader && !children) {
+      template = template ?? emptyStateDefaultTemplate;
+      const { iconType, pageTitle, description, rightSideItems } = pageHeader;
+      header = undefined;
+      children = (
+        <EuiEmptyPrompt
+          iconType={iconType}
+          iconColor={''} // This is likely a solution or app logo, so keep it multi-color
+          title={pageTitle ? <h1>{pageTitle}</h1> : undefined}
```

Comment:
We shouldn't do this inline.  Object and function creation in a prop leads to recreation on every render cycle.

```
const title = pageTitle ? <h1>{pageTitle}</h1> : undefined;
const body = description ? <p>{description}</p> : undefined;

children = (
        <EuiEmptyPrompt
          iconColor={''}
          actions={rightSideItems}
          {...{ iconType, title, body }}
        />
);
```